### ISSUE TITLE
[Feature Request]: Add report bug button to ingame escape menu.

### ISSUE THREAD

**Issue by CalebSerafin** [official-antistasi-community/A3-Antistasi#2013]
### Related problem

Players are generally too busy to fill out system or mission specifications.
By using a script that runs in the mission. It can pull all the relevant information (Map, Mods, Time, Is official version).
It can then automatically fill out fields of the issue template.

TL;DR
So all the player needs to do, is press a button and describe their problem.

### Proposed feature

Use URL queries to construct a bug report with important information already filled in. ****

We can even copy import logs to the clipboard for the player to type in.
The technology in the soon coming XML sanitiser, can be duplicated to perform URL sanitisation.

### Alternative solutions

It is do-able with current technology. No alternative is needed.

### Additional context

Visual Studio Code uses this feature to report bugs with extensions in two steps.
1. User Click on error. -> Opens GitHub URL. -> Page is already filled in with short OS and environment information.
2. User Paste in a large log.

**Comment by killerswin2** [comment]
Should this link to github, or the discord. I'm more in favor of discord.

**Comment by CalebSerafin** [comment]
> Should this link to GitHub, or the discord. I'm more in favour of discord.

Yeah. I guess to prevent issue spam and not force issue submitters to create GihHub accounts, these messages could be posted to a channel in Discord.

A viable solution is to host a relay in Azure Functions or something similar that forwards these bug reports to a discord webhook.

**Comment by Bob-Murphy** [comment]
We also have the possibility to direct people either to something hosted on our VM, our homepage and then have a bot/webhook forward it to the discord.
This way we can circumvent that people who might not have discord can still report something.

**Comment by rautamiekka** [comment]
Surely there's a 2-way Discord<->GitHub bot.

I just checked that unlike before when Discord allowed 30 threads per channel regardless of status, now it's 1000 **active (non-archived)** threads across the whole Server and **infinite archived** threads, which is just impossible to reach in a source code repo => such a bridge wouldn't be a bad idea at all.

**Comment by CalebSerafin** [comment]
Yeah, I do agree that the same flow could be extended to Discord to make it easier for the general public to create feedback and issues. And yes, it would be more cost-effective to host it on our existing VM infrastructure.

## For Creation
### From Arma
For making the request, we could do that totally in Antistasi with some edit boxes. On submission, it will be sent to GitHub.
### From Discord
We have a bot, so potentially that could also be modified to create GitHub issues when prompted by a user.

## Watching Activity and Staying Informed
On Discord, we use the basic GitHub webhook that posts updates on issues/PRs in a channel. That's usually sufficient for watching activities. If more info or interaction is required, it's probably fine requiring a GitHub account to do it.

## Risks
### Defacing
Using these semi-automated systems, a user from Discord (low likelihood) or Arma 3 (High likelihood) could attempt to post unsavoury issues. To mitigate this, the relay function should consult a user cooldown, user blacklist, and word filter database.

These basic features could form the Minimal Viable Product (MVP). After these, we could consider implementing editing and commenting from those original platforms. This basically allows the "full" experience of bug reports and feedback from the p[platform that the user is more experienced with.

**Comment by Bob-Murphy** [comment]
I wouldn't directly send it GitHub.
Whatever system we would have should first send it somewhere closed where it can be vetted and then confirmed/denied which then will send it to GitHub or not.
People will do stupid stuff either way and I rather not have 1000 issues with nonsense on GitHub.
At least that's what I would prefer. No clue if feasible.

Maybe a self hosted in-between system like https://github.com/Peppermint-Lab/peppermint or something similar could be a solution where issues could be checked first.
There are quite a lot of ticketing/helpdesk solutions available for self-hosting.

**Comment by CalebSerafin** [comment]
The plumming logic can be made in a single afternoon. 

Yeah, peppermint looks like a cool system. 
I will need to check that peppermint has an API to insert items and send off approved items. If not.my favourite trick is to just insert & read items directly into it's database.

If I were to develop it in-house: I would use an MVC application (server-side rendered web page), show a list of pending ticlets review. Clicking on a ticket shows a plain text or markdown render of the issue. 
With the view there are two buttons. One sends it to github with any applied edits. Other discards it (maybe keep storing them for 30 days incase of accident or evidence for blacklist).
I estimate that UI as two afternoons work.
